### PR TITLE
Add many "see also" links to docstrings

### PR DESCRIPTION
**Pull request by mcabbott**
This adds many more cross-links to the functions' docstrings. They are a little random, but the goal is to link to a few similar or contrasting functions whose existence, or precise names, might not be known.

They are mostly above `# Examples` as this seemed to be the policy. Although perhaps if the lists get longer, they ought to move to below.

I also edited a few other docstrings, including:
* giving examples of `floatmax` and `typemax` under each other's headings,
* ~~showing `similar` under `Array{T}(undef, ...)`~~
* mentioning that comprehensions use `collect` on various `Iterators.` objects.

Edit -- more:
* correcting `map` and `foreach` to say "iteration stops when any iterator is finished" and `map!` to say "`destination` must be at least as large as the smallest collection" not the first.
* adding examples to short-circuiting `&&` etc.
* adding examples like `ℯ^(im)π ≈ -1` to irrational constants.

I also added some functions to doc pages, to have something to refer to. Not sure how big a deal this is or isn't; I don't want to accidentally make policy.

### CODE REVIEW

**Comment by mcabbott** [comment]
CI doesn't like this, I'm not sure why. Is there a trick needed for these?
```
┌ Warning: no doc found for reference '[`@info`](@ref)' in src/base/base.md.
┌ Warning: no doc found for reference '[`copy!`](@ref)' in src/base/base.md.
┌ Warning: no doc found for reference '[`⊉`](@ref)' in src/base/collections.md.
```

**Comment by vtjnash** [comment]
Possibly not the canonical name for it? Often they are `[@ref Base.copy!]` and such

**Comment by fredrikekre** [comment]
Alternatively they are not included in the manual. For example `@info` does not exist, it is only documented as the generic [`@logmsg`](https://docs.julialang.org/en/v1/stdlib/Logging/#Logging.@logmsg)

**Comment by mcabbott** [comment]
Thanks for the hints. I'm slowly ironing them out locally, a few more builds to go.

File: base/iterators.jl [hunk 1]
```diff
@@ -925,6 +931,9 @@ julia> collect(Iterators.product(1:2, 3:5))
 2×3 Matrix{Tuple{Int64, Int64}}:
  (1, 3)  (1, 4)  (1, 5)
  (2, 3)  (2, 4)  (2, 5)
+
+julia> ans == [(x,y) for x in 1:2, y in 3:5]  # lowers to the same code
```
Comment (stevengj) [inline base/iterators.jl:935]: This is not true, I think.

While it lowers to very similar code (both use `Iterators.product`), it is not equivalent.  The comprehension version creates a `Generator` object, whereas collecting the iterator directly does not.

Comment (mcabbott) [reply base/iterators.jl:935]: Indeed that's not quite right, sorry, will fix. 

Literally showing what it is is probably too verbose, and not so easy to read... and will break `jldoctest` I presume. 
```
julia> ((x,y) for x in 1:2, y in 3:5)
Base.Generator{Base.Iterators.ProductIterator{Tuple{UnitRange{Int64}, UnitRange{Int64}}}, var"#65#66"}(var"#65#66"(), Base.Iterators.ProductIterator{Tuple{UnitRange{Int64}, UnitRange{Int64}}}((1:2, 3:5)))
```

**Comment by mcabbott** [comment]
I couldn't make a reference to `@logmsg` or `Logging.@logmsg`, but could make a reference to `Logging` which might be narrow enough. 

I am still not very clear on how these references work:
* How are they namespaced? Can I include references from Base to functions in standard libraries somehow, or is that impossible?
* Can I refer to labels like `kw"?"` for the ternary operator, or `kw"&&"`, from other docstrings? I think I deleted my attempts from the present state of this.

Maybe `@edit` here is the first problem, it's in `InteractiveUtils`.
```
��� Warning: no doc found for reference '[`Base.@edit`](@ref)' in src/base/base.md.
��� Warning: no doc found for reference '[`Base.@which`](@ref)' in src/base/base.md.
��� Warning: no doc found for reference '[`LinearAlgebra`](@ref)' in src/base/multi-threading.md.
��� Warning: no doc found for reference '[`Base.splat`](@ref)' in src/base/iterators.md.
��� Warning: 'Printf' is not unique in src/base/io-network.md.
��� Warning: reference for 'issubset' could not be found in src/base/collections.md.
```

**Comment by ViralBShah** [comment]
It would be great to get this in.

**Comment by mcabbott** [comment]
Excellent, what needs to happen, do you think? 

* Some functions are newly documented, i.e. added to `doc/src/base/...`. Maybe that's a review priority, does this make them more official? I'm not so sure of the policy here. 
* I'm still unclear how to create some cross-links, if anyone can enlighten me on https://github.com/JuliaLang/julia/pull/38606#issuecomment-749275839 I'd appreciate it. 
* More generally if anyone wants to scan a few of these and abject to any (or any patterns) that would also be great, even if you don't feel like reading all of them.

**Comment by ViralBShah** [comment]
I would think that the newly documented ones should be in a separate PR than the cross-links. That might perhaps make it easier to get things merged rather than all in one shot. 

I wonder what others have to say about that.

**Comment by mcabbott** [comment]
I don't mind moving them, although note that the two PRs will still be coupled, in the sense that links here won't work until their targets exist. The advantage of reviewing them in here is that you can easily search for the (proposed) links toward a given function. They should all be gathered [about here](https://github.com/JuliaLang/julia/pull/38606/files#diff-2edf8c1015857a22fdbd6c684a976bebb2baf23e39bc9d81f28f8f188807a978) on the super-long "files changed" page.

They are just things (like `promote_typejoin`) which seemed to deserve mention from somewhere (like `promote_type`) and cross-links without targets don't work. I have no strong opinions as to whether these functions are otherwise important enough to deserve documentation. If they don't, then the options are either to remove them entirely from "See also" to keep them hidden, or to mention them without making a hyperlink.

**Comment by ViralBShah** [comment]
Makes sense.

File: base/math.jl [hunk 2]
```diff
@@ -465,6 +479,8 @@ log(x::Number)
 Compute the logarithm of `x` to base 2. Throws [`DomainError`](@ref) for negative
 [`Real`](@ref) arguments.
 
+See also: [`exp2`](@ref), [`ldexp`](@ref).
+
```
Comment (vtjnash) [inline base/math.jl:483]: Note, in the merge, I lost the following corresponding update to `exp2`, when that doc moved from here to special/exp.jl:

```md
See also [`ldexp`](@ref), [`<<`](@ref).
```

File: base/abstractarray.jl [hunk 3]
```diff
@@ -374,6 +393,8 @@ first(a::AbstractArray) = a[first(eachindex(a))]
 Get the first element of an iterable collection. Return the start point of an
 [`AbstractRange`](@ref) even if it is empty.
 
+See also: [`only`](@ref), [`firstindex`](@ref), [`last`](@ref), [`tail`](@ref Base.tail).
```
Comment (vtjnash) [inline base/abstractarray.jl:396]:
```suggestion
See also: [`only`](@ref), [`firstindex`](@ref), [`last`](@ref).
```

I wouldn't generally consider tail to be similar. We could mention `popfirst!`, but I'm also not sure that is helpful.

File: base/abstractarray.jl [hunk 4]
```diff
@@ -2130,18 +2166,28 @@ end
...
-49
+julia> foreach(x -> push!(res, x^2), tri)
+
+julia> res
+3-element Vector{$(Int)}:
+  1
+ 16
+ 49
+
+julia> foreach((x,y) -> println(x," & ",y), tri, 1:100)
```
Comment (vtjnash) [inline base/abstractarray.jl:2187]:
```suggestion
julia> foreach((x, y) -> println(x, " & ", y), tri, 1:100)
```

File: base/file.jl [hunk 5]
```diff
@@ -730,8 +738,12 @@ end
 Apply the function `f` to the result of [`mktempdir(parent; prefix)`](@ref) and remove the
 temporary directory all of its contents upon completion.
 
+<<<<<<< HEAD
 !!! compat "Julia 1.2"
     The `prefix` keyword argument was added in Julia 1.2.
+=======
+See also: [`mktemp`](@ref), [`mkdir`](@ref).
+>>>>>>> another batch of see also additions
```
Comment (mcabbott) [inline base/file.jl:746]: An error I just spotted:
```suggestion
!!! compat "Julia 1.2"
    The `prefix` keyword argument was added in Julia 1.2.

See also: [`mktemp`](@ref), [`mkdir`](@ref).
```

File: base/abstractarray.jl (hunk 4 above)
Comment (mcabbott) [reply base/abstractarray.jl:2187]: Thanks for taking on the merge, I meant to have a go... 

Here, it's possible that writing something which is more obviously not code would be better, maybe `" and "`

File: base/array.jl [hunk 6]
```diff
@@ -1144,6 +1162,8 @@ Remove an item in `collection` and return it. If `collection` is an
 ordered container, the last item is returned; for unordered containers,
 an arbitrary element is returned.
 
+See also: [`popat!`](@ref), [`delete!`](@ref), [`splice!`](@ref), [`push!`](@ref).
```
Comment (vtjnash) [inline base/array.jl:1165]:
```suggestion
See also: [`popfirst!`](@ref), [`popat!`](@ref), [`delete!`](@ref), [`deleteat!`](@ref), [`splice!`](@ref), and [`push!`](@ref).
```

File: base/abstractarray.jl (hunk 3 above)
Comment (mcabbott) [reply base/abstractarray.jl:396]: Not so much similar, as often used together, was my thinking -- things like `g(x) = (f(first(x)), map(g, tail(x))...)`. But you may not know whether it's called `rest` or `allbutfirst` or something.

File: base/complex.jl [hunk 7]
```diff
@@ -7,6 +7,8 @@ Complex number type with real and imaginary part of type `T`.
 
 `ComplexF16`, `ComplexF32` and `ComplexF64` are aliases for
 `Complex{Float16}`, `Complex{Float32}` and `Complex{Float64}` respectively.
+
+See also: [`Real`](@ref), [`complex`](@ref), [`real`](@ref).
```
Comment (vtjnash) [inline base/complex.jl:11]: Seems like we weren't typically using a `:`, except this PR (which is inconsistent about it)? I think I'd suggest removing them all from this PR. Other opinions?

Comment (mcabbott) [reply base/complex.jl:11]: I'm not completely consistent, but my thinking was that some (esp. of the old ones) are more part of the discussion, "See also X which does Y, and Z.", so should be punctuated as sentences. While some are just a list of links, "See also: X, Y, Z."

File: base/docs/basedocs.jl [hunk 8]
```diff
@@ -692,7 +692,7 @@ See the manual section on [control flow](@ref man-conditional-evaluation) for mo
 ```
 julia> x = 1; y = 2;
 
-julia> println(x > y ? "x is larger" : "y is larger")
+julia> x > y ? println("x is larger") : println("y is larger")
```
Comment (vtjnash) [inline base/docs/basedocs.jl:695]: why this change? I thought the old version was more canonical

File: base/docs/basedocs.jl [hunk 9]
```diff
@@ -1816,10 +1846,18 @@ Symbol(x...)
...
-julia> tuple(1, 'a', pi)
-(1, 'a', π)
+julia> tuple(1, 'b', pi)
+(1, 'b', π)
+
+julia> ans === (1, 'b', π)
+true
+
+julia> Tuple([1, 2, pi])  # takes a collection
+(1.0, 2.0, 3.141592653589793)
```
Comment (vtjnash) [inline base/docs/basedocs.jl:1860]: I think this may be confusing, because of the promotion happening in the array.
```suggestion
julia> Tuple(Real[1, 2, pi])  # takes a collection
(1, 2, π)
```

File: base/docs/basedocs.jl (hunk 8 above)
Comment (mcabbott) [reply base/docs/basedocs.jl:695]: In the old one, there is nothing to run in the two expressions. `ifelse` would return the same answer.

File: base/docs/basedocs.jl [hunk 10]
```diff
@@ -1938,7 +1978,9 @@ isdefined
 """
     Vector{T}(undef, n)
 
-Construct an uninitialized [`Vector{T}`](@ref) of length `n`. See [`undef`](@ref).
+Construct an uninitialized [`Vector{T}`](@ref) of length `n`.
+
+See also [`undef`](@ref), [`similar`](@ref).
```
Comment (vtjnash) [inline base/docs/basedocs.jl:1983]: I think this change would be better suited for the primary docstring of ```Vector{T} <: AbstractVector{T}``` (for see also `similar`, `fill`, and `zeros`)

File: base/docs/basedocs.jl [hunk 11]
```diff
@@ -1988,14 +2030,21 @@ Vector{T}(::Missing, n)
 """
     Matrix{T}(undef, m, n)
 
-Construct an uninitialized [`Matrix{T}`](@ref) of size `m`×`n`. See [`undef`](@ref).
+Construct an uninitialized [`Matrix{T}`](@ref) of size `m`×`n`.
+
+See also [`undef`](@ref), [`similar`](@ref).
```
Comment (vtjnash) [inline base/docs/basedocs.jl:2035]: Same here as Vector

File: base/docs/basedocs.jl [hunk 12]
```diff
@@ -2043,19 +2092,29 @@ containing elements of type `T`. `N` can either be supplied explicitly,
 as in `Array{T,N}(undef, dims)`, or be determined by the length or number of `dims`.
 `dims` may be a tuple or a series of integer arguments corresponding to the lengths
 in each dimension. If the rank `N` is supplied explicitly, then it must
-match the length or number of `dims`. See [`undef`](@ref).
+match the length or number of `dims`.
+
+See also: [`undef`](@ref), [`similar`](@ref).
```
Comment (vtjnash) [inline base/docs/basedocs.jl:2097]: Same here as Vector and Matrix

File: base/docs/basedocs.jl (hunk 10 above)
Comment (vtjnash) [reply base/docs/basedocs.jl:1983]: and `empty`

File: base/essentials.jl [hunk 13]
```diff
@@ -131,7 +131,7 @@ end
 
 Tests whether variable `s` is defined in the current scope.
 
-See also [`isdefined`](@ref).
+See also [`isdefined`](@ref), [`isassigned`](@ref).
```
Comment (vtjnash) [inline base/essentials.jl:134]:
```suggestion
See also [`isdefined`](@ref) for field properties and [`isassigned`](@ref) for
array indexes or [`haskey`](@ref) for other mappings.
```

File: base/file.jl (hunk 5 above)
Comment (vtjnash) [inline base/file.jl:746]:
```suggestion
See also: [`mktemp`](@ref), [`mkdir`](@ref).

!!! compat "Julia 1.2"
    The `prefix` keyword argument was added in Julia 1.2.
```

File: base/irrationals.jl [hunk 14]
```diff
@@ -22,7 +22,9 @@ abstract type AbstractIrrational <: Real end
     Irrational{sym} <: AbstractIrrational
```
Comment (vtjnash) [inline base/irrationals.jl:22]: Instead of this see also link, how about we update all of these in another PR to  take the form:
```suggestion
    Irrational{sym} <: [`AbstractIrrational`](@ref)
```

File: base/iterators.jl [hunk 15]
```diff
@@ -544,6 +554,8 @@ rest(itr) = itr
 
 Returns the first element and an iterator over the remaining elements.
 
+See also: [`Iterators.drop`](@ref), [`Iterators.take`](@ref), [`Base.tail`](@ref).
```
Comment (vtjnash) [inline base/iterators.jl:557]:
```suggestion
See also: [`Iterators.drop`](@ref), [`Iterators.take`](@ref).
```

File: base/iterators.jl [hunk 16]
```diff
@@ -907,12 +925,17 @@ Return an iterator over the product of several iterators. Each generated element
 a tuple whose `i`th element comes from the `i`th argument iterator. The first iterator
 changes the fastest.
 
+See also: [`zip`](@ref), [`Iterators.flatten`](@ref), `Base.splat`.
```
Comment (vtjnash) [inline base/iterators.jl:928]:
```suggestion
See also: [`zip`](@ref), [`Iterators.flatten`](@ref), [`Base.splat`](@ref).
```

File: base/math.jl [hunk 17]
```diff
@@ -47,6 +47,8 @@ end
 Return `x` if `lo <= x <= hi`. If `x > hi`, return `hi`. If `x < lo`, return `lo`. Arguments
 are promoted to a common type.
 
+See also [`clamp!`](@ref), [`trunc`](@ref).
```
Comment (vtjnash) [inline base/math.jl:50]:
```suggestion
See also [`clamp!`](@ref).
```

Then for the `clamp(x, T)::T` docstring, add:
```julia
See also [`trunc(T, x)`](@ref).
```

File: base/iterators.jl (hunk 16 above)
Comment (mcabbott) [reply base/iterators.jl:928]: I think splat isn't yet in the manual. But perhaps it can be added, alongside `Base.:(|>)` and `Base.:(∘)`.

File: base/mathconstants.jl [hunk 18]
```diff
@@ -23,10 +23,17 @@ Base.@irrational catalan  0.91596559417721901505  catalan
 
 The constant pi.
 
+Unicode `π` can be typed by tab-completing `\\pi` in the Julia REPL.
```
Comment (vtjnash) [inline base/mathconstants.jl:26]:
```suggestion
Unicode `π` can be typed by writing `\\pi` then pressing tab in the Julia REPL, and in many editors.
```

File: base/mathconstants.jl [hunk 19]
```diff
@@ -37,10 +44,20 @@ julia> pi
 
 The constant ℯ.
 
+Unicode `ℯ` can be typed by tab-completing `\\euler` in the Julia REPL.
```
Comment (vtjnash) [inline base/mathconstants.jl:47]:
```suggestion
Unicode `ℯ` can be typed by writing `\\euler` and pressing tab in the Julia REPL, and in many editors.
```

File: base/number.jl [hunk 20]
```diff
@@ -110,6 +112,8 @@ copy(x::Number) = x # some code treats numbers as collection-like
 
 Returns `true` if the value of the sign of `x` is negative, otherwise `false`.
 
+See also [`sign`](@ref).
```
Comment (vtjnash) [inline base/number.jl:115]:
```suggestion
See also [`sign`](@ref) and [`copysign`](@ref).
```

File: base/operators.jl [hunk 21]
```diff
@@ -992,7 +1006,7 @@ julia> fs = [
 julia> ∘(fs...)(3)
 3.0
 ```
-See also [`ComposedFunction`](@ref).
+See also [`ComposedFunction`](@ref), [`!`](@ref).
```
Comment (vtjnash) [inline base/operators.jl:1009]:
```suggestion
See also [`ComposedFunction`](@ref), [`!f::Function`](@ref).
```

File: base/permuteddimsarray.jl [hunk 22]
```diff
@@ -83,10 +83,10 @@ end
 """
     permutedims(A::AbstractArray, perm)
 
-Permute the dimensions of array `A`. `perm` is a vector specifying a permutation of length
-`ndims(A)`.
+Permute the dimensions of array `A`. `perm` is a vector or a tuple of length `ndims(A)`,
```
Comment (vtjnash) [inline base/permuteddimsarray.jl:86]:
```suggestion
Permute the dimensions of array `A`. `perm` is a vector or a tuple of length `ndims(A)`
```

File: base/promotion.jl [hunk 23]
```diff
@@ -144,6 +143,17 @@ end
 Compute a type that contains both `T` and `S`, which could be
 either a parent of both types, or a `Union` if appropriate.
 Falls back to [`typejoin`](@ref).
+
+See also: [`promote`](@ref), [`promote_type`](@ref).
```
Comment (vtjnash) [inline base/promotion.jl:147]:
```suggestion
See instead [`promote`](@ref), [`promote_type`](@ref).
```

File: base/strings/io.jl [hunk 24]
```diff
@@ -57,6 +57,8 @@ end
 Print (using [`print`](@ref)) `xs` followed by a newline.
 If `io` is not supplied, prints to [`stdout`](@ref).
 
+See also: [`show`](@ref), [`@show`](@ref), [`@info`](@ref Logging), [`sprint`](@ref), [`Printf.@printf`](@ref man-printf).
```
Comment (vtjnash) [inline base/strings/io.jl:60]:
```suggestion
```

I don't think any of these are really that related to println just because they are alternatives to IO

File: base/tuple.jl [hunk 25]
```diff
@@ -110,6 +110,8 @@ in assignments, like `a, b... = collection`.
 !!! compat "Julia 1.6"
     `Base.rest` requires at least Julia 1.6.
 
+See also: [`first`](@ref first), [`tail`](@ref Base.tail), [`Iterators.rest`](@ref).
```
Comment (vtjnash) [inline base/tuple.jl:113]:
```suggestion
See also: [`first`](@ref first), [`Iterators.rest`](@ref).
```

File: stdlib/Printf/docs/src/index.md [hunk 26]
```diff
@@ -1,4 +1,4 @@
-# Printf
+# [Printf](@id man-printf)
```
Comment (vtjnash) [inline stdlib/Printf/docs/src/index.md:1]:
```suggestion
# Printf
```

This can already be referenced by name (e.g. `(Printf)[@ref]`)

**Comment by vtjnash** [comment]
Some comments of places to fix, but otherwise, lgtm

File: base/abstractarray.jl (hunk 3 above)
Comment (vtjnash) [reply base/abstractarray.jl:396]: `popfirst!` would be more typical for that need (or perhaps `Base.rest`)

File: base/iterators.jl (hunk 16 above)
Comment (vtjnash) [reply base/iterators.jl:928]: Oh, then it shouldn't be in the "see also" list (since you can't "see" it). Seems that it got added as a helper in 43ce8999207, so was not exported

File: base/complex.jl (hunk 7 above)
Comment (vtjnash) [reply base/complex.jl:11]: Yes, I think removing makes it more consistent for cases where a sentence is better than a simple list, and thus is overall better to use always. Let's see how the voting goes: https://github.com/JuliaLang/julia/pull/40401

File: base/iterators.jl (hunk 16 above)
Comment (mcabbott) [reply base/iterators.jl:928]: Sure, I can remove it. 

You can't click the link on the website, but at the repl there is no obvious distinction of what is and isn't documented. Maybe whether there should be is another question. 

Note that this PR does add some other functions to docs, discussed above: https://github.com/JuliaLang/julia/pull/38606#issuecomment-765269902 . It would seem stranger for me that `promote_type` not to cross link to `promote_typejoin`, since understanding that there are two ways in which Base widens seems important for anyone looking this up.

Comment (vtjnash) [reply base/iterators.jl:928]: Right. I reviewed everything, and thought it looked good, so I'm not suggesting changing any others based on this thread.

Comment (mcabbott) [reply base/iterators.jl:928]:
```suggestion
See also: [`zip`](@ref), [`Iterators.flatten`](@ref).
```

File: base/strings/io.jl (hunk 24 above)
Comment (mcabbott) [reply base/strings/io.jl:60]: This is too many, but perhaps a few things people who know the name of `println` might be wish they knew about? 
```suggestion
See also: [`@show`](@ref), [`@info`](@ref Logging), [`Printf.@printf`](@ref Printf).
```

File: base/docs/basedocs.jl (hunk 10 above)
Comment (mcabbott) [reply base/docs/basedocs.jl]: Done in https://github.com/JuliaLang/julia/pull/38606/commits/5f4284b63873e3942a7f79a27bb705417cce5f69

File: base/math.jl (hunk 17 above)
Comment (mcabbott) [reply base/math.jl]: Done in https://github.com/JuliaLang/julia/pull/38606/commits/273cd205f66934ad470b308259f705c79cf953d8, slightly enlarged...

File: base/irrationals.jl (hunk 14 above)
Comment (mcabbott) [reply base/irrationals.jl:22]: Does this work? On a toy example, the REPL prints the link out verbatim, rather than just highlighting the text. 

If it could be made to work, though, it would be good to have globally.

File: stdlib/Printf/docs/src/index.md (hunk 26 above)
Comment (mcabbott) [reply stdlib/Printf/docs/src/index.md]: Doing so give me this error:
```
��� Warning: 'Printf' is not unique in src/base/io-network.md.
��� @ Documenter.CrossReferences /buildworker/worker/package_linux64/build/doc/deps/packages/Documenter/bFHi4/src/CrossReferences.jl:100
```

File: base/docs/basedocs.jl (hunk 10 above)
Comment (mcabbott) [reply base/docs/basedocs.jl]: I think `empty` was not previously in docs. Added in https://github.com/JuliaLang/julia/pull/38606/commits/924f0fc13e18d3426ee07eab340216923d563f65 but I am not very sure that `doc/src/base/arrays.md ` is the right place, since it applies to many collections.

File: stdlib/Printf/docs/src/index.md (hunk 26 above)
Comment (mortenpi) [reply stdlib/Printf/docs/src/index.md:1]: There's a ["Printf" section on the NEWS page](https://docs.julialang.org/en/v1/NEWS/#Printf) as well, which is why it's not unique I believe.

File: base/irrationals.jl (hunk 14 above)
Comment (vtjnash) [reply base/irrationals.jl:22]: oh right, the indentation makes it a code block, and code blocks can't have links in them

We would need to go through and convert them all into code fences such as:
```suggestion
`Irrational{sym}` <: [`AbstractIrrational`](@ref)
```

File: base/strings/io.jl [hunk 27]
```diff
@@ -57,17 +57,20 @@ end
 Print (using [`print`](@ref)) `xs` followed by a newline.
 If `io` is not supplied, prints to [`stdout`](@ref).
 
+See also [`@show`](@ref) to print variables names with values,
+and [`Printf.@printf`](@ref man-printf) to format numbers.
+
```
Comment (vtjnash) [inline base/strings/io.jl:62]:
```suggestion
```

I still don't think these are relevant to `println`

File: base/math.jl (hunk 2 above)
Comment (vtjnash) [reply base/math.jl:502]: bump

**Review by vtjnash — COMMENTED** [review]
A couple nits, and we should merge this

File: stdlib/Printf/docs/src/index.md (hunk 26 above)
Comment (mcabbott) [reply stdlib/Printf/docs/src/index.md:1]: OK, the link is gone, but I leave the anchor, since it does no harm and may save someone figuring this out again some day.

File: base/math.jl (hunk 2 above)
Comment (mcabbott) [reply base/math.jl:502]: Now restored, sorry! Also added link from `exp(x)`

File: base/irrationals.jl [hunk 28]
```diff
@@ -19,7 +19,7 @@ for integers `n` will give a rational result when `n` is a perfect square), then
 abstract type AbstractIrrational <: Real end
 
 """
-    Irrational{sym} <: AbstractIrrational
+`Irrational{sym}` <: [`AbstractIrrational`](@ref)
```
Comment (vtjnash) [inline base/irrationals.jl:22]: oh, I didn't necessarily mean we should do this now. I'm probably in favor, but we should let others respond first in a separate PR

Comment (mcabbott) [reply base/irrationals.jl]: Sorry, now reverted. 

Possibly it could be done at a higher level, too, just make all types in docstrings clickable somehow?
